### ISSUE TITLE
"Uncaught ReferenceError: RTCPeerConnection is not defined"

### ISSUE THREAD

**Issue by yiannis-st** [AgoraIO/Basic-Video-Call#360]
Agora web : in my angular 14 application, bootstrap of the app will stop because of an error, in some devices. Mostly xiaomi devices and miBrowser  but not only will end up in a blank screen. Angular app will not be able to load, because the error will stop execution;

Related errors: 
"Uncaught ReferenceError: RTCPeerConnection is not defined"
"ReferenceError: Can't find variable: RTCPeerConnection"

![MicrosoftTeams-image (1)](https://user-images.githubusercontent.com/131011025/233589894-f3b14e6e-c43d-4fe1-b57e-11de00c6a084.png)
![MicrosoftTeams-image (2)](https://user-images.githubusercontent.com/131011025/233589901-0a5adc55-a346-4950-ba3a-3801be10bf10.png)

**Comment by plutoless** [comment]
@yiannis-st to run RTC feature, the browser needs to support webrtc. if RTCPeerConneciton is not defined, it usually means your browser does not support webrtc.
you may use checkSystemRequirements api to test the compatibility of current browser.

**Comment by yiannis-st** [comment]
The error is being thrown way before checkSystemRequirements are available.

**Comment by yiannis-st** [comment]
> @yiannis-st to run RTC feature, the browser needs to support webrtc. if RTCPeerConneciton is not defined, it usually means your browser does not support webrtc. you may use checkSystemRequirements api to test the compatibility of current browser.

Even if the webrtc is not supported, this do not mean that agora error should stop execution of the web page right?

**Comment by plutoless** [comment]
@yiannis-st correct. sorry for the inconvenience caused, maybe you can use a try catch to catch the error for now. I've raised an internal jira to engineer team
